TTS/auto_tts/model_hub.py

```diff
@@ -0,0 +1,543 @@
+            epochs=self.epochs,
+            lr=self.learning_rate,
+            memory_size=-1,
+            prenet_type="original",
+            use_forward_attn=forward_attn,
+            prenet_dropout=True,
+            attention_type="dynamic_convolution",
+            location_attn=location_attn,
+            attention_norm="sigmoid",
+            r=2,
+            stopnet=True,
+            separate_stopnet=True,
+            print_step=25,
+            plot_step=100,
+            print_eval=False,
+            save_step=10000,
+            checkpoint=True,
+            text_cleaner="phoneme_cleaners",
+            num_loader_workers=4,
+            num_eval_loader_workers=4,
+            batch_group_size=4,
+            min_seq_len=6,
+            max_seq_len=180,
+            compute_input_seq_cache=True,
+            output_path=self.output_path,
+            phoneme_cache_path=os.path.join(self.output_path, "phoneme_cache"),
+            use_phonemes=False,
+            phoneme_language="en-us",
+            datasets=[dataset],
+        )
+        return config
+
+    def ljspeech_glow_tts(self, audio, dataset, encoder):
+        if encoder == "transformer":
+            encoder_type = "rel_pos_transformer"
+        elif encoder == "gated":
+            encoder_type = "gated_conv"
+        elif encoder == "residual_bn":
+            encoder_type = "residual_conv_bn"
+        elif encoder == "time_depth":
+            encoder_type = "time_depth_separable"
+        elif encoder is None:
+            encoder_type = "rel_pos_transformer"
+
+        config = GlowTTSConfig(
+            audio=audio,
+            batch_size=self.batch_size,
+            eval_batch_size=self.batch_size // 2,
+            num_loader_workers=4,
+            num_eval_loader_workers=4,
+            run_eval=True,
+            test_delay_epochs=-1,
+            epochs=self.epochs,
+            text_cleaner="english_cleaners",
+            use_phonemes=False,
+            phoneme_language="en-us",
+            phoneme_cache_path=os.path.join(self.output_path, "phoneme_cache"),
+            print_step=25,
+            print_eval=True,
+            mixed_precision=False,
+            add_blank=False,
+            hidden_channels_enc=192,
+            hidden_channels_dec=192,
+            hidden_channels_dp=256,
+            use_encoder_prenet=True,
+            encoder_type=encoder_type,
+            r=1,
+            loss_masking=True,
+            grad_clip=5.0,
+            save_step=5000,
+            batch_group_size=0,
+            min_seq_len=3,
+            max_seq_len=500,
+            compute_f0=False,
+            use_noise_augment=True,
+            compute_input_seq_cache=True,
+            output_path=self.output_path,
+            datasets=[dataset],
+        )
+        return config
+
+    def ljspeech_speedy_speech(self, audio, dataset):
+        """Base speedy speech model for ljpseech dataset."""
+        model_args = SpeedySpeechArgs(
```

Comment:
SpeedySpeech is tricky to train since it needs character durations precomputed. You either compute them externally or train a Tacotron model first to compute durations. Maybe Speedy Speecy training should first start with the Tacotron training and compute the durations. But also it sounds like a lot of clutter in the code.

Reply:
Yea I saw that when testing that out, for now I'm just going to leave it out until I create a stable and clean experiment function